src/border.jl

```diff
@@ -652,13 +652,11 @@ The command `padarray(A,Fill(0,(1,1,1)))` yields
 ---
 
 """
-padarray(img::AbstractArray, border::Pad)  = padarray(eltype(img), img, border)
-function padarray(::Type{T}, img::AbstractArray, border::Pad) where T
-    inds = padindices(img, border)
-    # like img[inds...] except that we can control the element type
-    newinds = map(Base.axes1, inds)
-    dest = similar(img, T, newinds)
-    copydata!(dest, img, inds)
+padarray(img::AbstractArray, border::AbstractBorder)  = padarray(eltype(img), img, border)
+function padarray(::Type{T}, img::AbstractArray, border) where {T}
+    ba = BorderArray(img, border)
+    out = similar(ba, T)
+    copy!(out, ba)
```

Comment:
Can you do some performance comparisons between the existing implementation and this one? Your performance optimizations for `_copy!` are encouraging, of course, but just want to check.

Reply:
Yes I did. The current implementation is basically stolen from `padarray` anyway. I tinkered a bit with making it simpler or using the generic `copy`, but sadly performance was not as good.